mlir/tools/rocmlir-gen/rocmlir-gen.cpp

```diff
@@ -1429,60 +1535,67 @@ createCPUConvFunc(ModuleOp module,
   return func;
 }
 
-std::string getTypeStr(const mlir::Type &type) {
-  std::string typeName;
-  llvm::raw_string_ostream nameStream(typeName);
-  nameStream << type;
-  return nameStream.str();
-}
-
 static func::FuncOp getMemcpyFuncDecl(ModuleOp &module,
-                                      const mlir::Type &srcElemType,
-                                      const mlir::Type &dstElemType) {
+                                      const mlir::MemRefType srcType,
+                                      const mlir::MemRefType destType) {
   OpBuilder b(module.getContext());
 
-  // memcpy_<srcElemType>_<dstElemType>
+  assert(srcType.getRank() == 1 && "Memcopy takes 1-D sources");
+  assert(destType.getRank() == 1 && "Memcopy takes 1-D destinations");
+
+  mlir::Type srcElemType = srcType.getElementType();
+  mlir::Type dstElemType = destType.getElementType();
+  // memcpy_<srcElemType>_<dstElemType>_(srcSize|any)
   std::string funcName = "_memcpy_";
-  funcName += getTypeStr(srcElemType);
-  funcName += "_";
-  funcName += getTypeStr(dstElemType);
+  llvm::raw_string_ostream funcNameStr(funcName);
+  funcNameStr << srcElemType << "_" << dstElemType << "_";
```

Comment:
What is the benefit of using `raw_string_ostream`?

Reply:
Aesthetics, and possibly some performance gains from not using `std::string::operator+=`